pkg/kubectl/cmd/logs.go

```diff
@@ -199,7 +219,23 @@ func (o LogsOptions) Validate() error {
 
 // RunLogs retrieves a pod log
 func (o LogsOptions) RunLogs() (int64, error) {
-	req, err := o.LogsForObject(o.Object, o.Options)
+	switch t := o.Object.(type) {
+	case *api.PodList:
+		cnt := int64(0)
+		for _, p := range t.Items {
+			fmt.Fprintf(o.Out, "======== %s\n", p.Name)
```

Comment:
Yes it is a separator between pods.
With multiple containers it fails however silently. I was debating whether it was more correct to print the current error and continue or stop on the first error. Right now I don't even print the error which is worse.

With a slight fix, you will see the error message saying to specify a container name.
Obviously if you have multiple pods with different containers, the -l option isn't going to work well.
But I can provide more detail in the separator when you specify a container name.